src/lib/index.js

```diff
@@ -1182,31 +1182,30 @@ lib.isHidden = function(gd) {
     return !display || display === 'none';
 };
 
-lib.getTextTransform = function(opts) {
+lib.getTextTransform = function(opts, isCenter) {
```

Comment:
Why not place the `isCenter` option inside the `opts` object? I'm not a fan of generalising existing functions by adding boolean positional arguments, they make it hard to understand what the functions does when just looking at the caller.

---

Otherwise, thanks very much for cleaning this up. I see that you got a few more traces to reuse this routine - nice job!

Reply:
... oh can you add a jsDoc comment on top of this function? I have a feeling we'll be reusing this function in the future, so it would be nice to document its arguments and behaviour.

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/4420/commits/a27dc29ca87303fd53bf994bfe1203b3c5e96887.

Reply:
Ok. Now could you add info about all the different keys that the function expect in the `transform` object? For example, it would be nice to shed light on difference between `textX` and `targetX`

Reply:
Done in https://github.com/plotly/plotly.js/pull/4420/commits/1479943d15bb8fa09e7a3fa913ed2f0d576ea345

Reply:
Thanks!